### PR TITLE
Vision p6 pooling

### PR DESCRIPTION
**Pull request by psharath1**
BUG= Added support for pooling on Vision P6

### CODE REVIEW

File: tensorflow/lite/micro/kernels/xtensa/xtensa_pooling.h [hunk 1]
```diff
@@ -0,0 +1,49 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/xtensa_pooling.h:1]: 2022

File: tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+  }
+
+  micro_context->DeallocateTempTfLiteTensor(output);
+  micro_context->DeallocateTempTfLiteTensor(input);
+
+  return kTfLiteOk;
+}
+
+TfLiteStatus AvgPoolingPrepareVision(TfLiteContext* context, TfLiteNode* node) {
+  return (PoolingPrepareVision(context, node, AVG_POOLING));
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc:93]: Can drop the outer parens.

File: tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+}
+
+TfLiteStatus MaxPoolingPrepareVision(TfLiteContext* context, TfLiteNode* node) {
+  return (PoolingPrepareVision(context, node, MAX_POOLING));
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc:97]: Can drop the outer parens here too.

File: tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc [hunk 4]
```diff
@@ -0,0 +1,115 @@
+/* Copyright 2021 The TensorFlow Authors. All Rights Reserved.
```
Comment (rascani) [inline tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc:1]: 2022

**Review by rascani — CHANGES_REQUESTED** [review]
Overall LGTM

File: tensorflow/lite/micro/kernels/xtensa/xtensa_pooling.h (hunk 1 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/xtensa_pooling.h]: Updated

File: tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc (hunk 2 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc]: updated

File: tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc (hunk 3 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc]: updated

File: tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc (hunk 4 above)
Comment (psharath1) [reply tensorflow/lite/micro/kernels/xtensa/pooling_vision.cc]: Updated

**Review by tingyan19 — DISMISSED** [review]

**Review by rascani — DISMISSED** [review]

**Review by tingyan19 — APPROVED** [review]

**Pull request merged**